src/Stripe.net/Infrastructure/StringUtils.cs

```diff
@@ -52,5 +55,25 @@ public static bool SecureEquals(string a, string b)
 
             return result == 0;
         }
+
+        /// <summary>
+        /// Validates that the API key doesn't contain any whitespace characters.
+        /// </summary>
+        /// <param name="apiKey">The API key.</param>
+        /// <returns>The API key, if valid.</returns>
+        /// <exception name="StripeException">
+        /// Thrown if the API key contains whitespace characters.
+        /// </exception>
+        public static string ValidateApiKey(string apiKey)
+        {
+            if (!apiKeyRegex.IsMatch(apiKey ?? string.Empty))
+            {
+                var message = $"Invalid API key: \"{apiKey}\". API keys may not contain whitespace "
```

Comment:
Does an empty string pass? Sorry I don't really understand this if but the tests below imply empty string or null value works. Can you clarify why we would want that? Is it to explicitly unset the key?

Reply:
Both empty strings and `null` pass the validation. We need to allow `null` because it's the default value for strings.

That said, I've added another check in `StripeRequest` to ensure that we _do_ have a non-null, non-empty API key when creating the `Authorization` header that will be used in the actual HTTP request.